bridge/src/worker.rs

```diff
@@ -46,80 +38,54 @@ pub enum Response {
 pub type WorkerResult = Result<Vec<u8>, WorkerError>;
 type WorkerCallback = Box<dyn FnOnce(WorkerResult) + Send + 'static>;
 
-async fn handle_request(request: Request, outer_channel: SyncSender<Response>) {
-    match request {
-        Request::PollActivityTask { core_worker, callback } => {
-            let task = core_worker.poll_activity_task().await;
+async fn poll_activity_task_async(worker: Worker, callback: WorkerCallback) {
+    let task = worker.core_worker.poll_activity_task().await;
 
-            let response = match task {
-                Ok(task) => {
-                    let mut bytes: Vec<u8> = Vec::with_capacity(task.encoded_len());
-                    task.encode(&mut bytes).expect("Unable to encode activity task protobuf");
+    let response = match task {
+        Ok(task) => {
+            let mut bytes: Vec<u8> = Vec::with_capacity(task.encoded_len());
+            task.encode(&mut bytes).expect("Unable to encode activity task protobuf");
 
-                    Response::ActivityTask { bytes, callback }
-                },
-                Err(e) => {
-                    Response::Error { error: WorkerError::UnableToPollActivityTask(e), callback }
-                }
-            };
-
-            outer_channel.send(response).expect("Unable to send a callback");
+            Response::ActivityTask { bytes, callback }
         },
-    }
-}
-
-fn run_reactor_loop(runtime: Arc<Runtime>, mut receiver: tokio_mpsc::UnboundedReceiver<Request>, outer_sender: SyncSender<Response>) {
-    runtime.block_on(async move {
-        while let Some(request) = receiver.recv().await {
-            tokio::spawn(handle_request(request, outer_sender.clone()));
+        Err(e) => {
+            Response::Error { error: WorkerError::UnableToPollActivityTask(e), callback }
         }
-    });
-}
-
-fn reactor(runtime: Arc<Runtime>, outer_sender: SyncSender<Response>) -> &'static tokio_mpsc::UnboundedSender<Request> {
-    static REACTOR: OnceCell<tokio_mpsc::UnboundedSender<Request>> = OnceCell::new();
-    REACTOR.get_or_init(|| {
-        let (tx, rx) = tokio_mpsc::unbounded_channel::<Request>();
+    };
 
-        std::thread::spawn(move || run_reactor_loop(runtime, rx, outer_sender));
-
-        tx
-    })
+    worker.callback_tx.send(response).expect("Unable to send a callback");
 }
 
 #[derive(Clone)]
```

Comment:
I don't think you ever need to clone this entire worker (though it's mostly harmless). You can just clone the pieces you need (especially if you change to what I mentioned before about putting that top-level poll call's contents into the spawn).